db/in_memory_stats_history.h

```diff
@@ -12,6 +12,16 @@
 
 namespace rocksdb {
 
+// InMemoryStatsHistoryIterator can be used to access stats history that was
+// stored by an in-memory two level std::map(DBImpl::stats_history_). It keeps
+// a copy of the stats snapshot (in stats_map_) that is currently being pointed
+// to, which allows the iterator to access the stats snapshot even when
+// the background garbage collecting thread purges it from the source of truth
+// (`DBImpl::stats_history_`). In that case, the iterator will continue to be
+// valid until a call to `Next()` returns no result and invalidates it. In
+// some extreme cases, the iterator may also return fragmented segments of
+// stats snapshots due to long gaps between `Next()` calls and interleaved
+// garbage collection.
```

Comment:
It is a little bit confusing to me who this comment is for. Is it for developer or RocksDB, or is it for the user? If it is the former, I feel that a better home would be implementation of Next(), closer to the logic. If it is the latter, this should be somewhere in the public header, because this file is not available to them.